src/couch_replicator_httpc.erl

```diff
@@ -251,16 +258,41 @@ clean_mailbox(_, Count) when Count > 0 ->
 maybe_retry(Error, Worker, #httpdb{retries = 0} = HttpDb, Params) ->
     report_error(Worker, HttpDb, Params, {error, Error});
 
+%% For 429 errors, we perform an exponential backoff up to 250 * 2^15
+%% times, or roughly 2.17 hours. Since the #httpd.retries is initialized
+%% to 10 and we need 15, we use the Wait time as a timeout/failure end.
+maybe_retry(backoff, Worker, #httpdb{wait = Wait} = HttpDb, Params) ->
```

Comment:
I think we want this function clause at the top. If not, a job which is sitting at 0 retries will be killed permanently by a 429. Not a huge deal, but I suspect we want to keep the old semantics.